contracts/USDCFlow.cdc

```diff
@@ -18,159 +20,207 @@ import "FiatToken"
         ///
-        pub fun resolveView(_ view: Type): AnyStruct? {
-            switch view {
-                case Type<FungibleTokenMetadataViews.FTView>():
-                    return FungibleTokenMetadataViews.FTView(
-                        ftDisplay: self.resolveView(Type<FungibleTokenMetadataViews.FTDisplay>()) as! FungibleTokenMetadataViews.FTDisplay?,
-                        ftVaultData: self.resolveView(Type<FungibleTokenMetadataViews.FTVaultData>()) as! FungibleTokenMetadataViews.FTVaultData?
-                    )
-                case Type<FungibleTokenMetadataViews.FTDisplay>():
-                    let media = MetadataViews.Media(
-                            file: MetadataViews.HTTPFile(
-                            url: "https://assets.website-files.com/5f6294c0c7a8cdd643b1c820/5f6294c0c7a8cda55cb1c936_Flow_Wordmark.svg"
-                        ),
-                        mediaType: "image/svg+xml"
-                    )
-                    let medias = MetadataViews.Medias([media])
-                    return FungibleTokenMetadataViews.FTDisplay(
-                        name: "USDC (Flow)",
-                        symbol: "USDCf",
-                        description: "This fungible token representation of USDC is bridged from Flow EVM.",
-                        externalURL: MetadataViews.ExternalURL("https://www.circle.com/en/usdc"),
-                        logos: medias,
-                        socials: {
-                            "x": MetadataViews.ExternalURL("https://x.com/circle")
-                        }
-                    )
-                case Type<FungibleTokenMetadataViews.FTVaultData>():
-                    return FungibleTokenMetadataViews.FTVaultData(
-                        storagePath: USDCFlow.VaultStoragePath,
-                        receiverPath: USDCFlow.ReceiverPublicPath,
-                        metadataPath: USDCFlow.VaultPublicPath,
-                        providerPath: /private/usdcFlowVault,
-                        receiverLinkedType: Type<&USDCFlow.Vault{FungibleToken.Receiver}>(),
-                        metadataLinkedType: Type<&USDCFlow.Vault{FungibleToken.Balance, MetadataViews.Resolver}>(),
-                        providerLinkedType: Type<&USDCFlow.Vault{FungibleToken.Provider}>(),
-                        createEmptyVaultFunction: (fun (): @USDCFlow.Vault {
-                            return <-USDCFlow.createEmptyVault()
-                        })
-                    )
-            }
-            return nil
+        access(all) fun resolveView(_ view: Type): AnyStruct? {
+            return USDCFlow.resolveContractView(resourceType: nil, viewType: view)
         }
     }
 
-    /// createEmptyVault
-    ///
-    /// @return The new Vault resource with a balance of zero
-    ///
-    pub fun createEmptyVault(): @Vault {
-        return <-create Vault(balance: 0.0)
-    }
+    access(all) resource Minter: FlowEVMBridgeHandlerInterfaces.TokenMinter {
```

Comment:
I don't think it's an issue, but have we confirmed that adding the interface conformance in the 1.0 version migrates successfully?

Reply:
We're adding an entirely new resource type here, so there shouldn't be an interface conformance issue with this

Reply:
Ah right ofc